cleaner/custom_pre/classes/clean.php

```diff
@@ -0,0 +1,45 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * @package    cleaner_custom_pre
+ * @copyright  2019 Catalyst IT
+ * @author     Srdjan Janković <srdjan@catalyst.net.nz>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace cleaner_custom_pre;
+
+defined('MOODLE_INTERNAL') || die();
+
+class clean extends \local_datacleaner\clean {
+    const TASK = 'Custom tasks in pre pahse';
+
+    /**
+     * Execute custom tasks
+     */
+    static public function execute() {
+        global $DB;
+
+        $config = get_config('cleaner_custom_pre');
+
+        if (isset($config->sql)) {
+            $transaction = $DB->start_delegated_transaction();
```

Comment:
I'm curious why there is a transaction here, is this needed for any reason?

Also would this work if the sql had multiple statements in it? (which seems valid to me too)

Reply:
I thought there would be cases you would want it in one transaction. But can remove.

Multi statement works, there's a test for that.